File: main/src/main/java/com/sedmelluq/discord/lavaplayer/tools/io/EmptyInputStream.java
package com.sedmelluq.discord.lavaplayer.tools.io;

import java.io.InputStream;

/**
 * Represents an empty input stream.
 */
public class EmptyInputStream extends InputStream {
  public static final EmptyInputStream INSTANCE = new EmptyInputStream();

  @Override
  public int available() {
    return 0;
  }

  @Override
  public int read() {
    return -1;
  }
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/tools/io/EmptyInputStream.kt
package com.sedmelluq.discord.lavaplayer.tools.io

import java.io.InputStream

/**
 * Represents an empty input stream.
 */
class EmptyInputStream : InputStream() {
    companion object {
        @JvmField
        val INSTANCE = EmptyInputStream()
    }

    override fun available(): Int {
        return 0
    }

    override fun read(): Int {
        return -1
    }
}
